pkg/cloud/amazon/common.go

```diff
@@ -82,10 +126,22 @@ func GetCurrentlyConnectedRegionAndClusterName() (string, string, error) {
 	if err != nil {
 		return "", "", errors.Wrapf(err, "loading kubeconfig")
 	}
+
 	context := kube.Cluster(kubeConfig)
-	currentClusterName, currentRegion, err := ParseContext(context)
+	server := kube.CurrentServer(kubeConfig)
+	currentClusterName, err := GetClusterNameFromAWS("", "", server)
+	if err != nil {
+		currentClusterName, currentRegion, err := ParseContext(context)
```

Comment:
So the fallback is still ParseContext? Fair enough I guess.

Reply:
Yeah, i guess it might be a little redundant now though.